app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java

```diff
@@ -39,7 +42,9 @@ public void setListeners() {
         getView().findViewById(R.id.vLoginTermsAndConditionsText)
                 .setOnClickListener(it -> getPresenter().onTermsAndConditionsClicked());
         getView().findViewById(R.id.vLoginLoginButton)
-                .setOnClickListener(it -> getPresenter().onLoginClicked());
+                .setOnClickListener(it -> getPresenter()
+                        .onLoginClicked(((TextView) getView().findViewById(R.id.vLoginEmailField)).getText().toString(),
+                                        ((TextView) getView().findViewById(R.id.vLoginPasswordField)).getText().toString()));
```

Comment:
To make all of these better to read, you can define at the beginning of the class some private variables that represent you views. This is usually done like this:
```
TextView termsAndConditions = getView().findViewById(R.id.vLoginTermsAndConditionsText)
EditText email = getView().findViewById(R.id.vLoginEmailField)
```
Then you may be able to call your views like this:
```
email.setOnClickListener(it -> getPresenter().onLoginClicked( email.getText().toString() )
// I know you cast your editText emailField to (TextView) but EditText should have a getText() method by itself
```

Reply:
This can be applied to all your views, and you may get your code easy to read